File: backend/slotify/spring/src/main/java/com/slotify/backend/spring/company/dtos/GetCompanyResponse.java
package com.slotify.backend.spring.company.dtos;

import com.slotify.backend.spring.service.dtos.ServiceSummary;
import lombok.Builder;
import lombok.Getter;
import lombok.ToString;

import java.util.List;
import java.util.UUID;

@Builder
@Getter
@ToString
public class GetCompanyResponse {

    private UUID companyId;
    private Integer defaultMaxConcurrentServices;
    private String companyName;
    private String phoneNumber;
    private String emailAddress;
    private String physicalAddress;
    private String s3ImageKey;
    private String description;
    private Short rattingAvg;

    List<ServiceSummary> services;
}

File: frontend/app/src/main/java/com/raj/slotify/dtos/company/GetCompanyResponse.kt
package com.raj.slotify.dtos.company

import com.google.gson.annotations.SerializedName
import com.raj.slotify.dtos.service.ServiceSummary
import java.util.UUID

data class GetCompanyResponse(
    @SerializedName("companyId")
    val companyId: UUID,

    @SerializedName("defaultMaxConcurrentServices")
    val defaultMaxConcurrentServices: Int,

    @SerializedName("companyName")
    val companyName: String,

    @SerializedName("phoneNumber")
    val phoneNumber: String,

    @SerializedName("emailAddress")
    val emailAddress: String,

    @SerializedName("physicalAddress")
    val physicalAddress: String,

    @SerializedName("s3ImageKey")
    val s3ImageKey: String,

    @SerializedName("description")
    val description: String,

    @SerializedName("rattingAvg")
    val  rattingAvg: String,

    @SerializedName("services")
    val services: List<ServiceSummary>
)
